src/plugins/controls/server/options_list/options_list_queries.ts

```diff
@@ -139,6 +142,74 @@ const suggestionAggSubtypes: { [key: string]: OptionsListAggregationBuilder } =
       ),
   },
 
+  /**
+   * the "IP" query / parser should be used when the options list is built on a field of type IP.
+   */
+  ip: {
+    buildAggregation: ({ fieldName, searchString }: OptionsListRequestBody) => {
+      const ipSuggestions = {
+        terms: {
+          field: fieldName,
+          execution_hint: 'map',
+          shard_size: 10,
+        },
+      };
+
+      if (!searchString) {
+        return ipSuggestions;
+      }
```

Comment:
Instead of having this early return here when no search is provided, I **could** technically always do the range query and, when no IP is provided, the range would be from `0.0.0.0` to `255.255.255.255`. However, for the sake of efficiency, I figured it was better to introduce this third case although it makes the code slightly uglier 🤷